src/App.tsx

```diff
@@ -14,210 +53,115 @@ export const App: React.FC = () => {
     }
   }, []);
 
+  const visibleTodos = filterType === FilterType.All
+    ? todos
+    : todos.filter(({ completed }) => {
+      switch (filterType) {
+        case FilterType.Active:
+          return !completed;
+        case FilterType.Completed:
+          return completed;
+
+        default:
+          throw new Error();
+      }
+    });
+
+  const handleChooseFilter = useCallback(
+    (filter: FilterType) => {
+      setFilterType(filter);
+    },
+    [filterType],
+  );
+
+  const handleSubmit = useCallback(async (event: FormEvent) => {
+    event.preventDefault();
+
+    setIsAdding(true);
+
+    if (!title.trim()) {
+      setTodosError(TodosError.Title);
+      setTitle('');
+
+      return;
+    }
+
+    try {
+      if (!user) {
+        return;
+      }
+
+      const newTodo = await addTodos(user.id, title);
+
+      setTodos([...todos, newTodo]);
```

Comment:
If the new state is computed using the previous state, you can pass a function to setTodos